src/adt_simple.erl

```diff
@@ -86,31 +86,26 @@ server(S = #state { gossip = GS }) ->
 						)
 					);
 				% New block and confirmed txs callback.
-				{NewGS, {new_block, _, _, B, _, _}} ->
```

Comment:
An experiment would tell for sure, but from reading the code I conclude blocks accepted from peers are not sent to the internal gossip network - only the blocks mined by the node and the blocks received in the polling mode are.

If it's true, it can be fixed by calling `ar_gossip:send(GS, {new_block, ...`  somewhere [here](https://github.com/ArweaveTeam/arweave/blob/master/src/ar_node_utils.erl#L439), like it is done [here](https://github.com/ArweaveTeam/arweave/blob/master/src/ar_node_worker.erl#L599).

Reply:
I'm pretty sure it's the opposite that happens: the callback does get called when the node receives a block, but not if it asks for one by polling

Reply:
Ah, yes, it should be coming from [the bridge](https://github.com/ArweaveTeam/arweave/blob/master/src/ar_bridge.erl#L222). So as soon as it passes edge verification (timestamp, PoW) it is sent here. I guess we could live with it, but the module documentation could make it clearer, what do you think?